airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -2202,7 +2231,7 @@ def run_query(
         if encryption_configuration:
             configuration["query"]["destinationEncryptionConfiguration"] = encryption_configuration
 
-        job = self.insert_job(configuration=configuration, project_id=self.project_id)
+        job = self.insert_job(configuration=configuration, project_id=self.project_id, location=location)
```

Comment:
`insert_job()` will default to `self.location`

https://github.com/apache/airflow/blob/29fb38cfe3b0f1e54b48e7cc5786ab1c15ab8425/airflow/providers/google/cloud/hooks/bigquery.py#L1537

however, this change will break many tests